### PR TITLE
Explode archives in temp folder, Promote archiver package version to v3.2.2

### PR DESCRIPTION
**Pull request by barbelity**

### CODE REVIEW

File: utils/io/fileutils/files.go [hunk 1]
```diff
@@ -532,3 +533,37 @@ func IsEqualToLocalFile(localFilePath, md5, sha1 string) (bool, error) {
 	}
 	return localFileDetails.Checksum.Md5 == md5 && localFileDetails.Checksum.Sha1 == sha1, nil
 }
+
+// Copy directory content from one path to another.
```
Comment (eyalbe4) [inline utils/io/fileutils/files.go:537]: Copy --> Move

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
We should add tests for extractions of both zip and tar.gz formats.

File: httpclient/client.go [hunk 2]
```diff
@@ -606,48 +576,25 @@ func mergeChunks(chunksPaths []string, flags ConcurrentDownloadFlags) error {
...
-		if errorutils.CheckError(err) != nil {
-			return false, err
-		}
-		return true, nil
-	}
-
-	arch := archiver.MatchingFormat(flags.FileName)
-	if arch == nil {
+func handleArchive(flags ConcurrentDownloadFlags, logMsgPrefix string) error {
+	isSupportedArchive := fileutils.IsSupportedArchive(flags.FileName)
```
[the remark this answers is not in the document]
Comment (barbelity) [reply httpclient/client.go:580]: Removed the redundant check.

File: httpclient/client.go [hunk 3]
```diff
@@ -606,48 +576,25 @@ func mergeChunks(chunksPaths []string, flags ConcurrentDownloadFlags) error {
...
-	}
-
-	multiReader := io.MultiReader(fileReaders...)
 	extractionPath, err := getExtractionPath(flags.LocalPath)
 	if err != nil {
-		return false, err
+		return err
 	}
+
 	log.Info(logMsgPrefix+"Extracting archive:", flags.FileName, "to", extractionPath)
```
[the remark this answers is not in the document]
Comment (barbelity) [reply httpclient/client.go:591]: Removed the redundant log message.

File: utils/io/fileutils/files.go (hunk 1 above)
Comment (barbelity) [reply utils/io/fileutils/files.go:537]: Fixed.

**Pull request merged**